chat-widget/src/contexts/common/ILiveChatWidgetContext.ts

```diff
@@ -43,7 +44,9 @@ export interface ILiveChatWidgetContext {
         };
         e2vvEnabled: boolean; // true if voice/video calling is enabled and callingSDK instance created
         unreadMessageCount: number; // keep count of unread messages
-        conversationEndedByAgent: boolean; // true when agent ends the conversation
+        conversationEndedByAgentEventReceived: boolean; // true when agent end conversation or timeout event is received
```

Comment:
does this mean that in case of timeout, the widget will try to load the post-survey?, if that is the idea , would this backfire with the customer sending bad reviews due to the disconnection?

Reply:
This flag is solely set when we receive CustomerEndedThreadEvent from backend which could be because of agent end or disconnect. However this solely does not determine that we end chat. Since in case of bots this event is automatically fired after C2 ends chat. To handle these kind of scenarios we have a 2-step check where we set ConversationEndedBy Agent or Customer after running few logics against these events